pkg/source/mtadapter/adapter.go

```diff
@@ -183,54 +183,11 @@ func (a *Adapter) Update(ctx context.Context, obj *v1beta1.KafkaSource) error {
 		return nil
 	}
 
-	saslUser, err := a.ResolveSecret(ctx, obj.Namespace, obj.Spec.Net.SASL.User.SecretKeyRef)
+	kafkaEnvConfig, err := client.NewEnvConfigFromSpec(ctx, a.kubeClient, obj)
 	if err != nil {
 		return err
 	}
 
-	saslPassword, err := a.ResolveSecret(ctx, obj.Namespace, obj.Spec.Net.SASL.Password.SecretKeyRef)
-	if err != nil {
-		return err
-	}
-
-	saslType, err := a.ResolveSecret(ctx, obj.Namespace, obj.Spec.Net.SASL.Type.SecretKeyRef)
-	if err != nil {
-		return err
-	}
-
-	tlsCert, err := a.ResolveSecret(ctx, obj.Namespace, obj.Spec.Net.TLS.Cert.SecretKeyRef)
-	if err != nil {
-		return err
-	}
-
-	tlsKey, err := a.ResolveSecret(ctx, obj.Namespace, obj.Spec.Net.TLS.Key.SecretKeyRef)
-	if err != nil {
```

Comment:
isn't there a way to have the adapters mostly similar? Not sure I like they seem to diverge

Reply:
eg. have come "common adapter" ?

Reply:
the mtadapter is built on top of the stadapter so there is no real need for a common adapter. This code is not needed by the stadapter since secrets are resolved by k8s itself.